tensorflow_model_optimization/python/core/clustering/keras/cluster_test.py

```diff
@@ -503,5 +504,30 @@ def testStripSelectivelyClusteredSequentialModel(self):
     self.assertEqual(self._count_clustered_layers(stripped_model), 0)
     self.assertIsInstance(stripped_model.layers[0], layers.Dense)
 
+  @keras_parameterized.run_all_keras_modes
+  def testClusteringModelWithTrainableParamsLessNumberOfClusters(self):
+    """
+    Verifies that we skip a layer with the number of trainable parameters
+    less than the requested number of clusters.
+    """
+    model = keras.Sequential([
+        layers.Dense(2),
+        layers.Dense(1),
+        layers.Dense(4),
+        layers.Dense(3)   # 12 weights in kernel:0
+    ])
+    model.build(input_shape=(2, 1))
+
+    clustered_model = cluster.cluster_weights(model, **self.params)
+    self.assertTrue(not isinstance(clustered_model.layers[0], cluster_wrapper.ClusterWeights))
+    self.assertTrue(not isinstance(clustered_model.layers[1], cluster_wrapper.ClusterWeights))
+    self.assertTrue(not isinstance(clustered_model.layers[2], cluster_wrapper.ClusterWeights))
+    self.assertTrue(isinstance(clustered_model.layers[3], cluster_wrapper.ClusterWeights))
+
+    stripped_model = cluster.strip_clustering(clustered_model)
+
+    self.assertEqual(self._count_clustered_layers(stripped_model), 0)
+    self.assertEqual(model.get_config(), stripped_model.get_config())
```

Comment:
It may be good to check for the expected number of clusters in the clustered layer of the stripped model here too.

Reply:
Hi @akarmi Sorry, I think I don't understand which check to add. 
We don't properly train in this test: we just check that we don't wrap layers for clustering if they have the number of unique weights smaller than the requested number of clusters - so 3 layers are skipped, but the last one is wrapped. then we can do strip_clustering and remove what was added.